reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientTest.java

```diff
@@ -2489,4 +2493,89 @@ public void testLoopAndResolver() {
 		loop.disposeLater()
 		    .block(Duration.ofSeconds(30));
 	}
+
+	@Test
+	public void testCustomMetricsWithUriMapper() {
+		disposableServer =
+				HttpServer.create()
+							.port(0)
+							.handle((req, resp) -> resp.sendString(Mono.just("OK")))
+							.wiretap(true)
+							.bindNow();
+
+		Sinks.Many<String> collectedUris = Sinks.many().replay().all();
```

Comment:
I see little sense in using a Sink/Flux here, a `List` (make it a `CopyOnWriteArrayList` if you're concerned about concurrency) would be perfectly fine here.

Reply:
@simonbasle The PR used List, but with List it is not guaranteed that exactly after receiving the response  all values will be in the least (see previous commits and the failed CI). So we either use `Sinks` or other mechanism (CountDownLatch) in order to be sure that when we start checking the values all data is there.

We can use CoundDownLatch if this is your recommendation.

Reply:
I see. I would have expected the metrics to be recorded by the time the client`.get()` is finished (since it is blocking). I found the pattern of using `Sinks` a bit surprising and harder to read than a list-based assertion in that particular case.

Using `take(5)` is a potential subtle issue, too. It wouldn't catch a bug where a 6th uri is recorded.

So my preference would be to switch back to a `List`+`assertJ` approach. reactor-netty tests don't depend on `Awaitility` from what I can see, but it could be a better alternative than `CountdownLatch` here, wdyt @violetagg ?

Plus the assertion would be simpler and be less brittle:

```java
assertThat(collectedUris).hasSize(5).containsOnly("/stream/{n}");
```

Either way is fine in the end, I'll leave that decision to you @violetagg. I would just add a clarifying comment to the use of `Sinks` if we keep this solution.

Reply:
the number `5` is because we have 5 types of metrics and not 5 requests, because of that we cannot expect on `get` to have all of them.
For one request that we make we have `recordDataReceived`, `recordDataSent`, `recordDataReceivedTime`, `recordDataSentTime`, `recordResponseTime`

Reply:
yeah, I understood that. I meant, if a bug is introduced in the `MetricsRecorder` (or the internal code that calls it) and ends up calling eg. `recordDataReceived` twice, this test wouldn't see it (due to the `take(5)`) whereas a list assertion like I suggested would.

Reply:
@fkjellberg Please switch to `CountDownLatch` or `Awaitility` instead of `Sinks`